plugins/cython/compile_cython.py

```diff
@@ -37,43 +44,66 @@ def product_types(cls):
     def cache_target_dirs(self):
         return True
 
-    @property
-    def create_target_dirs(self):
-        return True
-
-    def __init__(self, context, workdir):
-        super(CompileCythonCreate, self).__init__(context, workdir)
-
     def execute_codegen(self, target, results_dir):
-        self.context.log.info("Processing target %s" % target)
-
-        # Creating the pex can, as a side effect, change the location of the symbolic link pants creates
-        result = self.create_pex(PexInfo.default())
-        full_path = os.path.join(get_buildroot(), target.target_base)
-        # subprocess.check_call(['cmake', full_path], cwd=results_dir)
-        # subprocess.check_call(['make'], cwd=results_dir)
-
-        result_code = result.run(
-            with_chroot=True,
-            blocking=True,
-            args=(os.path.join(full_path, 'setup.py'), 'build_ext', '--inplace', '--verbose'),
-            # Passing PATH helps cython find the correct c++ compiler
-            env={'libraries': results_dir, 'PATH': os.getenv('PATH')}
-        )
-
-        if result_code != 0:
-            raise ValueError('creating cython library failed')
-
-        library_source_path = os.path.join(result.path(), target.output)
-
-        library_output = os.path.join(results_dir, target.output)
-        safe_mkdir_for(library_output)
-        shutil.move(library_source_path, library_output)
-
-        self.context.log.info('created library {}'.format(os.path.relpath(target.output, get_buildroot())))
-
-    def synthetic_target_type_by_target(self, target):
-        return Resources
+        self.context.log.info("Processing target {}".format(target))
+
+        requirements_pex = self.context.products.get_data(ResolveRequirements.REQUIREMENTS_PEX)
+
+        interpreter = self.context.products.get_data(PythonInterpreter)
+        pex_info = PexInfo.default(interpreter)
+        pex_info.pex_path = requirements_pex.path()
+        with temporary_dir() as source_pex_chroot:
+            sources_pex_builder = PEXBuilder(
+                path=source_pex_chroot,
```

Comment:
This is the equivalent of using the `PEX_PATH` environment variable with pex (see pex --help-variables), but, in short, it allows us to logically union the `requirements_pex` (ie: cython) to the source code pex we build here. Keeping these two pexes seperate is alot of what `PythonExecutionTaskBase` was helping you do before and it is the engine behind performance wins in the new pipeline. It used to be invalidation of source files that went into the pex forced re-resolution of requirements. Now these two things invalidate separately.